src/ProviderPR/SFA.DAS.ProviderPR/Application/Requests/Commands/AddAccountRequestCommandHandler.cs

```diff
@@ -0,0 +1,134 @@
+﻿using MediatR;
+using SFA.DAS.ProviderPR.Common;
+using SFA.DAS.ProviderPR.Infrastructure;
+using SFA.DAS.ProviderPR.InnerApi.Notifications.Commands;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.InnerApi.Requests;
+using SFA.DAS.SharedOuterApi.InnerApi.Responses;
+using SFA.DAS.SharedOuterApi.Interfaces;
+using static SFA.DAS.SharedOuterApi.InnerApi.Responses.GetAccountTeamMembersWhichReceiveNotificationsResponse;
+
+namespace SFA.DAS.ProviderPR.Application.Requests.Commands;
+
+public enum Role { Owner }
+
+public class AddAccountRequestCommandHandler(
+    IProviderRelationshipsApiRestClient _providerRelationshipsApiRestClient,
+    IAccountsApiClient<AccountsConfiguration> _accountsApiClient
+) : IRequestHandler<AddAccountRequestCommand, AddAccountRequestCommandResult>
+{
+    public async Task<AddAccountRequestCommandResult> Handle(AddAccountRequestCommand command, CancellationToken cancellationToken)
+    {
+        var addAccountResponse = await _providerRelationshipsApiRestClient.CreateAddAccountRequest(command, cancellationToken);
+
+        var teamMembers = await _accountsApiClient.GetAll<TeamMember>(new GetAccountTeamMembersByInternalAccountIdRequest(command.AccountId));
+
+        if (!teamMembers.Any())
+        {
+            return new AddAccountRequestCommandResult(addAccountResponse.RequestId);
+        }
+
+        PostNotificationsCommand notificationCommand = new();
+
+        // If the provided EmployerContactEmail is null - then we must run through each account owner and send out the 'AddAccountOwnerInvitation' notification.
+
+        if (string.IsNullOrWhiteSpace(command.EmployerContactEmail))
+        {
+            CreateNotificationsForMembers(ref notificationCommand, command, teamMembers);
+
+            await SendNotifications(notificationCommand, cancellationToken);
```

Comment:
remove comment and call this SendNotificationsToAddAccountOwnerInvitation(

Reply:
@MarkFCain The send notifications method is called for all three Notification types and in some cases for multiple different types at the one time so has to be generic.